docs/src/pages/docs/crocks/Tuple.md

```diff
@@ -0,0 +1,480 @@
+      Sum.empty(), encodeCorrect(payload), First.empty()
+    )
+
+  case 'complete':
+    return Triple(
+      Sum.empty(), Sum.empty(), First(true)
+    )
+  }
+
+  return reportEmpty()
+}
+
+// extract :: ReportTriple -> Triple Number Number Boolean
+const extract =
+  trimap(valueOf, valueOf, option(false))
+
+// decode :: (Number, Number, Boolean) -> Object
+const decode = (attempts, correct, complete) =>
+  ({ attempts, correct, complete })
+
+// foldReport :: [ EventRecord ] -> ReportTriple
+const foldReport = mapReduce(
+  encode,
+  flip(concat),
+  reportEmpty()
+)
+
+// calculate :: [ EventRecord ] -> Object
+const calculate = compose(
+  merge(decode),
+  extract,
+  foldReport
+)
+
+calculate(data)
+//=> { attempts: 2, correct: 3, complete: true }
+```
+
+<article id="topic-implements">
+
+## Implements
+
+`Setoid`, `Semigroup`, `Functor`
+
+</article>
+
+<article id="topic-construction">
+
+## Construction
+
+```haskell
+Tuple :: Number -> n-Tuple *...n
+```
+
+`Tuple` is a type constructor that takes a non-zero, positive `Integer` as its
+argument. Instead of an instance, it will return a constructor that is used
+to construct an `n` sized `Tuple`, where `n` is the number of independent values
+that can be represented.
+
+The resulting `n-Tuple` constructor is parameterized as an `n-Functor`, where
+each parameter can vary from instance to instance.
+
+```javascript
+import Tuple from 'crocks/Tuple'
+
+// Triple :: 3-Tuple
+const Triple =
+  Tuple(3)
+
+// Quad :: 4-Tuple
+const Quad =
+  Tuple(4)
+
+Triple(false, true, 45)
+//=> 3-Tuple (false, true, 45)
+
+Quad({ a: true }, [ 1, 2, 3 ], 60, 'string')
+//=> 4-Tuple ({ a: true }, [ 1, 2, 3 ], 60, "string")
+```
+
+</article>
+
+<article id="topic-instance">
+
+## Instance Methods
+
+#### equals
+
+```haskell
+n-Tuple *...n ~> a -> Boolean
+```
+
+Used to compare the values of two `n-Tuple` instances by value.`equals` takes
+any given argument and will return a `true` if passed an `n-Tuple` of the same
+size with match values in the same positions as the `n-Tuple` `equals` was
+run on. If the provided argument is not an `n-Tuple` of the same type or the
+underlying values are not equal, then `equals` will return `false`.
+
+
+```javascript
+import Tuple from 'crocks/Tuple'
+
+// Pair :: 2-Tuple
+const Pair =
+  Tuple(2)
+
+// Triple :: 3-Tuple
+const Triple =
+  Tuple(3)
+
+Pair(1, false)
+  .equals(Pair(1, false))
+//=> true
+
+Triple(1, false, [ 1, 2 ])
+  .equals(Triple(1, false, [ 1, 2 ]))
+//=> true
+
+Triple(1, false, [ 1, 2 ])
+  .equals(Triple(1, true, [ 3, 4 ]))
+//=> false
+
+Pair(1, false)
+  .equals(Triple(1, false, [ 1, 2 ]))
+//=> false
+```
+
+#### concat
+
+```haskell
+Semigroup s1 => 1-Tuple s1 ~> 1-Tuple s1 -> 1-Tuple s1
+Semigroup s1, s2 => 2-Tuple s1 s2 ~> 2-Tuple s1 s2 -> 2-Tuple s1 s2
+...
+```
+
+When a given `n-Tuple` contains `Semigroup` instances in all defined `n`
+positions. It can be concatenated with another `n-Tuple` instance containing
```

Comment:
IMHO, little change but should be better to use a `,` instead of `.` here. It makes the read more easy.
> When a given `n-Tuple` contains `Semigroup` instances in all defined `n`
> positions, it can be concatenated with another `n-Tuple`...